### ISSUE TITLE
Changing parameters of Reverb (or other real-time enabled effect) influences sound during preview playback with a long lag, ~10s

### ISSUE THREAD

**Issue by dozzzzer** [audacity/audacity#2954]
**Describe the bug**
When the Reverb effect is called and playback is started, changing sliders position/parameter values won't affect the audio.

**Precondition**
Delete pluginregistry.cfg before the launch of the app to enable `EffectRealtime`

**To Reproduce**
1. Load an audio clip
2. Select a region and loop it
3. Go to Effect > Reverb (or Wahwah)
4. Preview play by clicking the **Start Playback** button and change the sliders position
5. Observe: the sliders have no effect on the sound
6. Observe: the **Start Playback** button does not change to **Stop Playback** 

**Actual behavior**
No changes to the sound.

**Expected behavior**
The sound is affected by the change of parameters of the effect.
**Start Playback** button changes to **Stop Playback**  during preview playback

**Screenshots**
As you can see from the video, none of the parameters I'm changing affect the sound in any way:
https://user-images.githubusercontent.com/70063970/169539071-7e70c582-0ad1-46aa-acd3-886884ad3904.mp4

**Additional information (please complete the following information):**
 - OS: MacOS Monterey
 - Version: [audacity-macOS-3.2.0-alpha-20220518+0f11289-universal](https://github.com/audacity/audacity/suites/6556705386/artifacts/245151879)

Discovered while testing #2923

**Comment by Paul-Licameli** [comment]
I think I have seen this.  I think:

- it is not special to Reverb but would be seen with other real-time capable effects
- There is an update of parameters, if you wait for it, but the lag can be many seconds
- There is a other play button on the effect dialog.  Does it behave like the main play button or differently?  Maybe a clue for debugging
- The fix might be a simple tuning of a few numbers governing the lengths of the RingBuffers

**Comment by petersampsonaudacity** [comment]
Testing on W10 with latest alpha master audacity-win-3.2.0-alpha-20220520+c07880c-x64-msvc2022

This appears to similarly afflict Wahwah - and this case it is a regression on 3.1.3, confirming @Paul-Licameli 's comment above

_**Note:** not a regression for Reverb as this effect was not real-time enabled in 3.1.3 and earlier_

**Comment by petersampsonaudacity** [comment]
This affects all real-time enable effects

a) Audacity's own:  Bass & Treble, Distortion, Phaser and Wahwah (all a regression on 3.1.3)

b) It also affects other Real-time enabled plug-ins - for me I see the same failure with iZotope Vinyl - and this is a regression on 3.1.3 and earlier

**Comment by Ryan-Miller-Audacity** [comment]
I noted this as well with AudioUnit effects on macOS, so I would say Peter is correct in that this effects all real-time enabled effects.

Including that checking/unchecking the "enable" box does not have does not have an effect on the realtime playback.

**Comment by Ryan-Miller-Audacity** [comment]
This issue is blocking full QA testing of making effects realtime (#2923).

**Comment by Paul-Licameli** [comment]
Observed at commit 0730e83a9dfdc1d010b3313a894eec6ed83eab2f:

- If you put Wahwah in the sidebar, start play, press the gear -- then the sound DOES adjust.
- If you have no sidebar effects, open the non-modal WahWah dialog, press the Play button OF THE TOOLBAR, and adjust sliders -- the sound DOES adjust as expected.
- If instead you start the play with the "Start Playback" button, then the sound DOES NOT adjust.

I also observe that "Start Playback" does change to "Stop Playback", with either way of starting the effect, unlike what the original report says.

So there is only some need now to explain and fix the third point.

**Comment by Paul-Licameli** [comment]
Update to that last comment:  In the third case, the sound DOES adjust -- but with a very long lag, 10 seconds.  Wait for it.

**Comment by pietro68** [comment]
I tried the current head `a3b3b53` and now I can see that all 4 points above test ok, and the regression I noticed before is now gone - all realtime changing of params seems to work.

**Comment by petersampsonaudacity** [comment]
Tested on W10 with latest master audacity-win-3.2.0-alpha-20220623+2fd21af-x64-msvc2022

@Paul-Licameli  This works OKish in that the sliders move and there is no lag for the changes 

**_BUT_** I do get nasty crackly/burbly artifacts as I move the sliders

**Comment by pietro68** [comment]
Regarding the issue with changing the reverb parameters in realtime: I just found it is really a problem specific to the reverb. I opened an issue for that at https://github.com/audacity/audacity/issues/3155.  Other issues mentioned in this issue were solved, so I am closing it.

**Comment by DeeFeeCee** [comment]
Had this happen to me while using MeldaProductions' MEqualizer. Was using for about an hour, honing in on the right frequencies, playing & pausing normally. Suddenly, I pressed space to play it again & it reverted my changes to some older state (I had not changed made any other changes). Now every time I go to start the song, it resets the harmonics & maximal count.

After some fiddling, I got it to retain the changes if I made them while the song was playing; now it's back to normal behavior. Not sure if this is relevant, but I had a portion of the track looped. Why it was fine in the first 90 minutes I was using it, bugged for 5 minutes, & is now working fine is beyond me, but either this plugin or realtime effects themselves are unstable.
